common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/LockingHandler.java

```diff
@@ -126,52 +128,69 @@ private void lookAtLockedTarget() {
      */
     private void bowAimingAssist() {
         LocalPlayer player = WorldUtils.getClientPlayer();
+        if (player == null) return;
+
+        // Check if player is using a bow
         if (config.aimAssistEnabled && !aimAssistActive && player.isUsingItem() && player.getUseItem().getItem() instanceof BowItem) {
             List<Entity> hostileEntities = BuiltinEntityPOIGroups.HOSTILE.group.getItems();
-            if (!hostileEntities.isEmpty()) {
-                Entity entity = hostileEntities.stream()
-                        .min(Comparator.comparingDouble(e -> WorldUtils.getClientPlayer().distanceTo(e)))
-                        .get();
-                if (lockOnEntity(entity)) {
-                    aimAssistActive = true;
+            if (hostileEntities != null && !hostileEntities.isEmpty()) {
+                Optional<Entity> closestEntity = hostileEntities.stream()
+                        .min(Comparator.comparingDouble(player::distanceTo));
+
+                if (closestEntity.isPresent()) {
+                    Entity entity = closestEntity.get();
+                    if (lockOnEntity(entity)) {
+                        aimAssistActive = true;
+                    }
                 }
             }
         }
 
-        if (aimAssistActive && !player.isUsingItem()) {
+        // Reset when not using bow anymore
+        if (aimAssistActive && (!player.isUsingItem() || !(player.getUseItem().getItem() instanceof BowItem))) {
             unlock(false);
             aimAssistActive = false;
             lastAimAssistCue = -1;
             lastBowState = -1;
+            return;
         }
 
-        if (config.aimAssistAudioCuesEnabled && aimAssistActive) {
+        // Handle audio cues for aiming
+        if (config.aimAssistAudioCuesEnabled && aimAssistActive && lockedOnEntity != null) {
             float bowPullingProgress = BowItem.getPowerForTime(player.getTicksUsingItem());
 
             int bowState = -1;
             if (bowPullingProgress >= 0f && bowPullingProgress < 0.50f) bowState = 0;
             if (bowPullingProgress >= 0.50f && bowPullingProgress < 1f) bowState = 1;
             if (bowPullingProgress == 1f) bowState = 2;
 
-            if (PlayerUtils.isPlayerCanSee(player.getEyePosition(), PlayerUtils.currentEntityLookingAtPosition, lockedOnEntity)) {
-                if (lastAimAssistCue != 1 || bowState != lastBowState) {
-                    PlayerUtils.playSoundOnPlayer(SoundEvents.NOTE_BLOCK_PLING, config.aimAssistAudioCuesVolume, bowState);
-                    lastAimAssistCue = 1;
+            Vec3 eyePosition = player.getEyePosition();
+            Vec3 targetPosition = PlayerUtils.currentEntityLookingAtPosition;
+
+            if (targetPosition != null) {
+                boolean canSeeTarget = PlayerUtils.isPlayerCanSee(eyePosition, targetPosition, lockedOnEntity);
+
+                if (canSeeTarget) {
```

Comment:
Why are we extracting this into a variable?
```suggestion
                if (PlayerUtils.isPlayerCanSee(eyePosition, targetPosition, lockedOnEntity)) {
```